source/common/http/conn_manager_impl.cc

```diff
@@ -218,15 +218,6 @@ void ConnectionManagerImpl::doEndStream(ActiveStream& stream) {
   }
 
   checkForDeferredClose();
-
-  // Reading may have been disabled for the non-multiplexing case, so enable it again.
-  // Also be sure to unwind any read-disable done by the prior downstream
-  // connection.
-  if (drain_state_ != DrainState::Closing && codec_->protocol() < Protocol::Http2) {
```

Comment:
This is a really exciting cleanup: less branching on protocol in HCM and less readDisable loops in odd places.  Thanks for doing this!

Reply:
+1 thanks this is great.